src/chat/ChatScreen.tsx

```diff
@@ -96,6 +99,24 @@ export const ChatScreen: React.FC<ChatScreenProps> = ({
     [firebaseInstance, memberIds, partners]
   );
 
+  const onLoadEarlier = useCallback(async () => {
+    if (isLoadingRef.current) {
+      return;
+    }
+    isLoadingRef.current = true;
+    if (conversationRef.current?.id) {
+      const res = await firebaseInstance.getMoreMessage();
+      setHasMoreMessages(res.length === 20); // if the length of the response is 20, it means there are more messages to load
+      setMessagesList((previousMessages) =>
+        GiftedChat.prepend(previousMessages, res)
+      );
+    }
+
+    setTimeout(() => {
+      isLoadingRef.current = false;
+    }, 300);
```

Comment:
why we need delay 300ms?

Reply:
yes, this need timeout too make sure not duplicated **onLoadEarlier** function in 300ms